boshio/boshio.go

```diff
@@ -194,7 +196,10 @@ func (c Client) retryableRequest(stemcellURL string, byteRange string) ([]byte,
 	byteRangeHeader := fmt.Sprintf("bytes=%s", byteRange)
 	req.Header.Add("Range", byteRangeHeader)
 
-Retry:
+	return c.retry(req)
+}
+
+func (c Client) retry(req *http.Request) ([]byte, error) {
```

Comment:
I dig the recursion but did we actually need two functions to accomplish this? Does the one `retryableRequest` function just get longer?